src/features/applications/components/ApplicationReviewActions/index.tsx

```diff
@@ -0,0 +1,42 @@
+"use client";
+
+import { useApplicationReview } from "../../hooks/useApplicationReview";
+import { ApplicationStatusBadge } from "./ApplicationStatusBadge";
+import { ReviewActionButton } from "./ReviewActionButton";
+
+type Props = {
+  projectId: string;
+};
+
+export function ApplicationReviewActions({ projectId }: Props) {
+  const {
+    status,
+    isAdmin,
+    userCanRevoke,
+    onApprove,
+    onRevoke,
+    revokeIsPending,
+    approveIsPending,
+  } = useApplicationReview({ projectId });
+
+  if (!status) return null;
+
+  if (!isAdmin) return <ApplicationStatusBadge status={status} />;
+
+  const type = status === "approved" ? "revoke" : "approve";
+  const isLoading = type === "approve" ? approveIsPending : revokeIsPending;
+  const disabled = type === "revoke" ? !userCanRevoke : false;
+  const onClick = type === "approve" ? onApprove : onRevoke;
```

Comment:
These definitions can probably be moved into the `ReviewActionButton` component. Just pass both `onApprove` and `onRevoke`.